README.md

```diff
@@ -101,36 +100,56 @@ Prerequisites:
     export OASIS_GO=go1.13.8
     ```
 
-* [Rust](https://www.rust-lang.org) and the nightly toolchain.
+* [Rust].
 
-  Once you have [`rustup` installed](https://www.rust-lang.org/tools/install),
-  install the nightly with:
+  We follow [Rust upstream's recommendation][rust-upstream-rustup] on using
+  [rustup] to install and manage Rust versions.
 
-  ```
-  rustup install nightly
-  ```
+  _NOTE: [rustup] cannot be installed alongside a distribution packaged [Rust]
+  version. You will need to remove it before you can start using [rustup]._
 
-  Then make it the default version with:
+  Install it by running:
 
   ```
-  rustup default nightly
+  curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
   ```
 
-* [Fortanix Rust SGX](https://edp.fortanix.com) target.
+  _NOTE: If you want to avoid directly executing a shell script fetched the
+  internet, you can also [download `rustup-init` executable for your platform]
+  and run it manually._
 
-  Install it by running:
+  This will run `rustup-init` which will download and install the latest stable
+  version of [Rust] on your system.
+
+  The version of the [Rust] toolchain we use in Oasis Core is specified in the
+  [rust-toolchain] file and will be downloaded and installed automatically.
+
+* [Fortanix Rust EDP] utilities.
+
+  Install them by running:
 
   ```
-  rustup target add x86_64-fortanix-unknown-sgx
+  cargo install fortanix-sgx-tools sgxs-tools
   ```
 
-  Install its utilities by running:
+  _NOTE: These utilities must be compiled with a nightly version of the [Rust]
+  toolchain since they use the `#![feature]` macro.
+  Make sure you run the above command from the Oasis Core git checkout or by
+  making a nighty version of the [Rust] toolchain the default in [rustup]._
+
+* Fortanix SGX [Rust] target.
+
+  Make sure you are in Oasis Core git checkout directory so that [rustup] will
+  add the Fortanix SGX [Rust] target to the [Rust] toolchain specified in
+  Oasis Core's [rust-toolchain] file.
+
+  Then install it by running:
 
   ```
-  cargo install fortanix-sgx-tools sgxs-tools
+  rustup target add x86_64-fortanix-unknown-sgx
```

Comment:
why are the `rustup target add` and `cargo install` reversed from before?

Reply:
you only need the target for compiling to sgx. the sgx tools are binaries that compile to native. You'd be best off using `rustup target add --toolchain nightly-2020-02-16 x86_64-fortanix-unknown-sgx` if you don't set the default toolchain during the initial install.

Reply:
As @nhynes explained, the Fortanix SGX target is not needed for compiling the sgx tools, however, sgx tools are required for compiling to sgx.

I would like to avoid hard-coding a particular nightly version in the README because experience suggests we will forget to update it.
Hence, I tried to use such instructions that never need to specify a nightly version and only rely on the `rust-toolchain` file.